src/Model/Post/Media.php

```diff
@@ -1178,6 +1178,12 @@ public static function getUrlForId(int $id, string $size = ''): string
 	 */
 	public static function getAllocatedHeightByMedia(array $media): string
 	{
-		return (100 * $media['height'] / $media['width']) . '%';
+		if (!empty($media['height']) && !empty($media['width'])) {
+			return (100 * $media['height'] / $media['width']) . '%';
+		}
+		if (!empty($media['preview-height']) && !empty($media['preview-width'])) {
```

Comment:
What’s the case where `width` and `height` aren’t filled but `preview-width` and `preview-height` are?

Reply:
I don't know. But I saw that where the function is called, we are using the preview and the none preview value as well: https://github.com/friendica/friendica/blob/dd7bea4bd146acf52086d303fce34a83d52f2516/src/Model/Item.php#L3364